pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_test.go

```diff
@@ -200,3 +202,255 @@ func TestDeployReconcileAsPlacementNotExist(t *testing.T) {
+			Timeout: clusterv1alpha1.Timeout{Timeout: "None"},
+		},
+	}
+	allRollOut := clusterv1alpha1.RolloutStrategy{
+		Type: clusterv1alpha1.All,
+		All: &clusterv1alpha1.RolloutAll{
+			Timeout: clusterv1alpha1.Timeout{Timeout: "None"},
+		},
+	}
+
+	mwrSet := helpertest.CreateTestManifestWorkReplicaSetWithRollOutStrategy("mwrSet-test", "default",
+		map[string]clusterv1alpha1.RolloutStrategy{placement1.Name: allRollOut, placement2.Name: perGoupeRollOut})
+
+	mw := helpertest.CreateTestManifestWork(mwrSet.Name, mwrSet.Namespace, placement2.Name, "cls1")
+	fWorkClient := fakeworkclient.NewSimpleClientset(mwrSet, mw)
+	workInformerFactory := workinformers.NewSharedInformerFactoryWithOptions(fWorkClient, 1*time.Second)
+
+	err = workInformerFactory.Work().V1().ManifestWorks().Informer().GetStore().Add(mw)
+	assert.Nil(t, err)
+
+	err = workInformerFactory.Work().V1alpha1().ManifestWorkReplicaSets().Informer().GetStore().Add(mwrSet)
+	assert.Nil(t, err)
+
+	mwLister := workInformerFactory.Work().V1().ManifestWorks().Lister()
+	pmwDeployController := deployReconciler{
+		workApplier:         workapplier.NewWorkApplierWithTypedClient(fWorkClient, mwLister),
+		manifestWorkLister:  mwLister,
+		placeDecisionLister: placementDecisionLister,
+		placementLister:     placementLister,
+	}
+
+	mwrSet, _, err = pmwDeployController.reconcile(context.TODO(), mwrSet)
+	assert.Nil(t, err)
+	assert.Equal(t, len(mwrSet.Status.PlacementsSummary), len(mwrSet.Spec.PlacementRefs))
+
+	// Check placements summary
+	expectPlcSummary1 := workapiv1alpha1.PlacementSummary{
+		Name: placement1.Name,
+		AvailableDecisionGroups: getAvailableDecisionGroupProgressMessage(len(placement1.Status.DecisionGroups),
+			int(placement1.Status.NumberOfSelectedClusters), placement1.Status.NumberOfSelectedClusters),
+		Summary: workapiv1alpha1.ManifestWorkReplicaSetSummary{
+			Total:       int(placement1.Status.NumberOfSelectedClusters),
+			Applied:     0,
+			Available:   0,
+			Degraded:    0,
+			Progressing: 0,
+		},
+	}
+	expectPlcSummary2 := workapiv1alpha1.PlacementSummary{
+		Name: placement2.Name,
+		AvailableDecisionGroups: getAvailableDecisionGroupProgressMessage(len(placement2.Status.DecisionGroups),
+			clsPerGroup, placement2.Status.NumberOfSelectedClusters),
+		Summary: workapiv1alpha1.ManifestWorkReplicaSetSummary{
+			Total:       clsPerGroup,
+			Applied:     0,
+			Available:   0,
+			Degraded:    0,
+			Progressing: 0,
+		},
+	}
+	for _, plcSummary := range mwrSet.Status.PlacementsSummary {
+		if plcSummary.Name == placement1.Name {
+			assert.Equal(t, plcSummary, expectPlcSummary1)
+		} else if plcSummary.Name == placement2.Name {
+			assert.Equal(t, plcSummary, expectPlcSummary2)
+		} else {
+			t.Fatal("PlacementSummary Name not Exist")
+		}
+	}
+
+	// Check for the expected manifestWorkReplicaSetSummary
+	mwrSetSummary := workapiv1alpha1.ManifestWorkReplicaSetSummary{
+		Total:       clsPerGroup + int(placement1.Status.NumberOfSelectedClusters),
```

Comment:
What if there's clusters overlap in placement1 and placement2, will the cluster number be counted twice in the summary?

Reply:
My thought is a cluster counted twice makes the meaning of [Total](https://github.com/open-cluster-management-io/api/blob/main/work/v1alpha1/types_manifestworkreplicaset.go#L116) confusing to the user, since the actual number of ManifestWorks is less than it. 

For addon, one addon only belongs to one placement (the latter one), and use that rollout strategy do rollout. Maybe it's not the same as the mwrSet here, just for reference